### PR TITLE
Add support for AWS postgresql standard log format

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

AWS PostgreSQL logs are separated with `:` instead of the supported `,` and `<space>` in https://github.com/elastic/integrations/blob/main/packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/default.yml#L24-L29. This PR is to create a new pipeline called `pipeline-aws-log.yml` and use `:` as separator instead.

### AWS PostgreSQL log format
```
%t:%r:%u@%d:[%p]:t
```
%t – Time of log entry.
%r – Remote host address.
%u@%d – User name @ database name.
[%p] – Process ID if available.
t - actual log message

Sample log:
```
2019-03-10 03:54:59 UTC:10.0.0.123(52834):postgres@tstdb:[20175]:ERROR: column "wrong" does not exist at character 8
```
From the log format you can see, AWS PostgreSQL logs are separated with `:` between different fields. 

Source: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.Concepts.PostgreSQL.html

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/integrations/issues/3540

### CODE REVIEW

**Comment by ravikesarwani** [comment]
Thanks Kaiyan for driving the fix to this issue.
@rameshelastic/services team let us know if there are any questions. Appreciate your help in getting it reviewed so that we can get this delivered and in the hands of cloud observability team.

File: packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+description: Pipeline for parsing PostgreSQL logs.
+processors:
+  - grok:
+      field: raw_message
+      ignore_missing: true
+      patterns:
+        - '(%{IPORHOST:source.ip}\(%{INT:source.port:int}\))?%{USERNAME:user.name}?@%{POSTGRESQL_DB_NAME:postgresql.log.database}?:(\[%{NUMBER:process.pid:long}\])?:%{WORD:log.level}: %{GREEDYDATA:message}'
```
Comment (tommyers-elastic) [inline packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml:8]: is the timestamp missing from this pattern, or am just reading this wrong?

Comment (tommyers-elastic) [inline packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml:8]: will the host always be in the form of `ip:port` ? or could it be a  domain name?

**Review by tommyers-elastic — COMMENTED** [review]
looking good - couple of questions though

File: packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml (hunk 1 above)
Comment (kaiyan-sheng) [reply packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml:8]: `timestamp` field is extracted before getting into this `pipeline-aws-log.yml` section. It's processed in `default.yml`.

Comment (kaiyan-sheng) [reply packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml:8]: Ha good point! I think it's actually possible. Let me read a bit more from the AWS documentation and add the change in the next commit. Thanks for the review!

Comment (kaiyan-sheng) [reply packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml]: done

**Review by girodav — APPROVED** [review]
LGTM :)

File: packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+processors:
+  - grok:
+      field: raw_message
+      ignore_missing: true
+      patterns:
+        - '(%{DATA:postgresql.log.client_addr}\(%{NUMBER:postgresql.log.client_port:int}\)|\[%{DATA:postgresql.log.client_addr}\])?:%{USERNAME:user.name}?@%{POSTGRESQL_DB_NAME:postgresql.log.database}?:(\[%{NUMBER:process.pid:long}\])?:%{WORD:log.level}: %{GREEDYDATA:message}'
+      pattern_definitions:
+        GREEDYDATA: |-
+          (.|
+          |	)*
```
Comment (ishleenk17) [inline packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml:12]: Would suggest to use escape sequences here to make it more readable and robust

Comment (kaiyan-sheng) [reply packages/postgresql/data_stream/log/elasticsearch/ingest_pipeline/pipeline-aws-log.yml]: Done, let me know what you think!

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**